msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/errors/ResetPasswordErrors.kt

```diff
@@ -0,0 +1,88 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+package com.microsoft.identity.nativeauth.statemachine.errors
+
+import com.microsoft.identity.nativeauth.statemachine.results.ResetPasswordResult
+import com.microsoft.identity.nativeauth.statemachine.results.ResetPasswordStartResult
+import com.microsoft.identity.nativeauth.statemachine.results.ResetPasswordSubmitPasswordResult
+
+/**
+ * ResetPasswordErrorTypes class holds the specific error type values that can be returned
+ * only by the reset password flow.
+ */
+internal class ResetPasswordErrorTypes {
+    companion object {
+        /*
+         * The PASSWORD_RESET_FAILED value indicates that the password reset flow failed.
+         */
+        const val PASSWORD_RESET_FAILED = "password_reset_failed"
+    }
+}
+
+/**
+ * Reset password error. The user should use the utility methods of this class
+ * to identify and handle the error. This error is produced by
+ * [com.microsoft.identity.client.INativeAuthPublicClientApplication.resetPassword]
+ * @param errorType the error type value of the error that occurred.
+ * @param error the error returned by the authentication server.
+ * @param errorMessage the error message returned by the authentication server.
+ * @param correlationId a unique identifier for the request that can help in diagnostics.
+ * @param errorCodes a list of specific error codes returned by the authentication server.
+ * @param exception an internal unexpected exception that happened.
+ */
+class ResetPasswordError(
```

Comment:
File name and class name should match. We should change the file name here.

Reply:
Under this file, there are multiple classes like ResetPasswordError and ResetPasswordSubmitPasswordError.  SignInErrors has SignInError, SignInUsingPasswordError, SignInSubmitPasswordError. Should we rename only one of them as the file name？

Reply:
yeah, that's true. I actually prefer "errors" because this file contains multiple classes, not just one. calling this file `ResetPasswordError.kt` would be incorrect. We could split this into multiple files, but that seems like slight overkill.
@SaurabhMSFT what lint/code analysis tool does this recommendation come from?

Reply:
yes. If this file didn't have `ResetPasswordErrorTypes` in it then the plural `ResetPasswordErrors.kt` would have been correct as all the remaining classes would be homogeneous. Due to the presence of `ResetPasswordErrorTypes` there are heterogeneous classes in this file and the name should be singular `ResetPasswordError.kt`.